presto-hudi/src/main/java/com/facebook/presto/hudi/HudiSplitManager.java

```diff
@@ -114,10 +116,10 @@ public ConnectorSplitSource getSplits(
         // Load Hudi metadata
         ExtendedFileSystem fs = getFileSystem(session, table);
         HoodieMetadataConfig metadataConfig = HoodieMetadataConfig.newBuilder().enable(isHudiMetadataTableEnabled(session)).build();
-        Configuration conf = fs.getConf();
+        StorageConfiguration<Configuration> conf = getStorageConfWithCopy(fs.getConf());
         HoodieTableMetaClient metaClient = HoodieTableMetaClient.builder().setConf(conf).setBasePath(table.getPath()).build();
         HoodieTimeline timeline = metaClient.getActiveTimeline().getCommitsTimeline().filterCompletedInstants();
-        String timestamp = timeline.lastInstant().map(HoodieInstant::getTimestamp).orElse(null);
+        String timestamp = timeline.lastInstant().map(HoodieInstant::requestedTime).orElse(null);
```

Comment:
Sth to audit to make sure that for table version 8 and above the latest instant based on the completion time should be returned.